.github/workflows/basic-tests-uv.yml

```diff
@@ -0,0 +1,66 @@
+name: Code tests (uv)
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - '**/*.py'
+      - '**/*.ipynb'
+      - '**/*.yaml'
+      - '**/*.yml'
+      - '**/*.sh'
+  pull_request:
+    branches: [ main ]
+    paths:
+      - '**/*.py'
+      - '**/*.ipynb'
+      - '**/*.yaml'
+      - '**/*.yml'
+      - '**/*.sh'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  uv-tests:
+    name: UV Tests (Matrix OS)
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python (uv)
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install uv and dependencies
+        shell: bash
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          uv python install 3.10
+          uv add . --dev
+          uv pip install -r ch05/07_gpt_to_llama/tests/test-requirements-extra.txt
+          uv add pytest-ruff nbval
```

Comment:
Is it important to use `uv add` here, or can you just do

```
uv pip install --system .
uv pip install --system -r ch05/07_gpt_to_llama/tests/test-requirements-extra.txt
uv pip install --system pytest-ruff nbval
```

and then skip manually activating the virtual environment for each step?

Given that you do this for `pip` as well I assume there's a reason(?).

Reply:
Ah, this is for testing purposes, too. 

In addition, `uv add . --dev` installs from the `pyproject.toml` and should be faster. I think `uv pip install --system .` may be slower as it uses the pip algo. 

The reason why I am doing 

```
uv pip install -r ch05/07_gpt_to_llama/tests/test-requirements-extra.txt
```

though is that this is bonus material 95% of the users don't care about, and so there is no pyproject.toml for this one.

Reply:
Hm, I don't think it is any faster. If anything it should be slower as it needs to open and edit a file (in the case of `uv add pytest-ruff nbval`) and then run `uv pip install <whatever your uv add target was>`. AFAIK, `uv` doesn't have multiple installer algorithms, it just has `uv pip install` and then higher level APIs on top of that. Would be interested if I'm wrong though.

Reply:
that's why I thought but someone told me yesterday that `uv add` is supposed to be faster as it doesn't try to mimic pip legacy behavior. Should check some official docs...

Reply:
Hm. The docs (https://docs.astral.sh/uv/reference/cli/#uv-add) don't mention that, but maybe @charliermarsh can comment on if there's interesting under the hood behavior.

Reply:
@matthewfeickert or maybe I misinterpreted the "Yes" [here](https://x.com/rasbt/status/1890860459131121720)

<img width="588" alt="Screenshot 2025-02-17 at 2 26 24 PM" src="https://github.com/user-attachments/assets/cd38eefe-e807-427d-9d39-e46e0bcf416e" />